### ISSUE TITLE
Sync of entity approval status  for encounter type and program should sync data linked to all subject types

### ISSUE THREAD

**Issue by ashusvnath** [avniproject/avni-server#565]
Analysis details:
Currently`fetchSubjectTypeForEntityNameAndUuid` used in EntityApprovalStatusController for getting entityApprovals, fetches only the first subject type linked to encounter type or program. This results in fetching entityApprovalStatus based on only the sync settings of the first subject type. This needs to be fixed.

**Comment by vinayvenu** [comment]
This is true for other entities as well, such as encounters that are linked to multiple subject types. We talked about this a while ago, and there was no clear answer on how we can handle this correctly. We might even have a card to track this. 

The subject type is important to fetch the address level types that their subject is linked to. We use these address level types to fetch the right address levels that should be used to filter by catchment. 

The issue occurs when an encounter type is linked to multiple subject types, and each of these subject types have a different level at which they are registered. 

This is a core issue in our sync, and requires deeper discussion to figure out a solution. 

cc @himeshr @petmongrels
